docs/user-guide/projects.md

```diff
@@ -54,6 +54,75 @@ argocd proj add-destination <PROJECT> <CLUSTER>,<NAMESPACE>
 argocd proj remove-destination <PROJECT> <CLUSTER>,<NAMESPACE>
 ```
 
+We can also do negations of destinations (i.e. install anywhere _apart from_).
+
+```bash
+argocd proj add-destination <PROJECT> !<CLUSTER>,!<NAMESPACE>
+argocd proj remove-destination <PROJECT> !<CLUSTER>,!<NAMESPACE>
+```
+
+Declaratively we could do something like this:
+
+```yaml
+spec:
+  destinations:
+  # Do not allow any app to be installed in `kube-system`  
+  - namespace: '!kube-system'
+    server: '*'
+  # Or any cluster that has a URL of `team1-*`   
+  - namespace: '*'
+    server: '!https://team1-*'
+    # Any other namespace or server is fine though.
```

Comment:
I think if we're going to provide this example which demonstrates an incorrect understanding of the `destinations` field (one that assumes order is significant), then we should loudly flag that this is an example of an **incorrect** `destinations` config.

Reply:
How about this:

> Declaratively we could do something like this, _in theory_:
> The default semantics of how a destination evaluates, when negation is involved, is undefined, so don't do it like below!!!

```yaml
spec:
  # This config is incorrect due to the fact that ordering is not defined, so please DON'T do this at home!!!
  destinations:
  # Do not allow any app to be installed in `kube-system`  
  - namespace: '!kube-system'
    server: '*'
  # Or any cluster that has a URL of `team1-*`   
  - namespace: '*'
    server: '!https://team1-*'
    # Any other namespace or server is fine though.
  - namespace: '*'
    server: '*'
```

Reply:
Yep, I like that. Should prevent the over-eager copy/paste. :-)